Zend/zend_operators.c

```diff
@@ -2503,11 +2516,23 @@ static void ZEND_FASTCALL increment_string(zval *str) /* {{{ */
 	int ch;
 
 	if (Z_STRLEN_P(str) == 0) {
+		zend_error(E_DEPRECATED, "Increment on non-alphanumeric string is deprecated");
+		if (EG(exception)) {
+			return;
+		}
 		zval_ptr_dtor_str(str);
 		ZVAL_CHAR(str, '1');
 		return;
 	}
 
+	if (UNEXPECTED(!zend_string_only_has_ascii_alphanumeric(Z_STR_P(str)))) {
```

Comment:
I think I did it that way because part of the problem that we are modifying the string in place.

So a string ``aa zz`` would be modified to ``aa aa`` then emit the deprecation notice, if a userland handler converts this to an exception, the initial string *should* still be ``aa zz`` IMHO

Reply:
Oh, right. Never mind then.